students/clifford_butler/lesson03/mailroom.py

```diff
@@ -0,0 +1,162 @@
+from a menu of 3 actions: “Send a Thank You”, 
+“Create a Report” or “quit”.
+'''
+
+import sys
+
+donor_dict = [
+        ["William Gates, III", 653772.32, 12.17],
+        ["Jeff Bezos", 877.33],
+        ["Paul Allen", 663.23, 43.87, 1.32],
+        ["Mark Zuckerberg", 1663.23, 4300.87, 10432.0],
+        ["Alexandra Butler", 777.77, 44.44],
+        ]
+
+prompt = "\n".join(("Welcome to the mail room!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Quit",
+          ">>> "))
+
+prompt_name = "\n".join(("Type the donors full name or,",
+              "type 'list' to display a list of the donors names.",
+              ">>> "))
+
+prompt_amount = "\n".join(("Whats the donation amount?",
+                ">>> "))
+
+exit_report = "\n".join(("Press 1 to exit to the initial prompt.",
+                         "\n"))
+
+report_header = "\n".join(("Donor Name           | Total Given | Num Gifts | Average Gift",
+                            "{:-^61}")).format('')
+
+def dis_info():
+    for item in donor_dict:
+        spaces = ((20 - len(item[0]))*" ")
+        
+        sum_donations = sum(item[1:])
+        sum_donations = round(sum_donations,2)
+        num_gifts = len(item[1:])
+        avg_gift = (str(round(sum_donations/num_gifts,2)))
+        spaces_two = ((15 - len(str(sum_donations)))*' ')
+        spaces_three = (42*" ")
+        
+        print(item[0], spaces, ('$'), sum_donations, spaces_two, num_gifts, spaces_three, ('$'), avg_gift)
+
+def get_index(donor_name):
+    # Return the index number based on user input
+    for item in (donor_dict):
+        if item[0] == donor_name:
+            index = donor_dict.index(item)
+            
+            return index
+        
+    return None
+
+def add_donor(donor_name):
+    # Add donor name to the data structure.
+    for item in donor_dict:
+        if item[0] not in donor_dict:
+            donor_dict.append([donor_name])
+            break
+
+def view_list():
+    # Display the donor list.
+    print("\n".join(donor_dict))
+
+def send_thank_you():
+    # Request the user to input donor name, and donation information.
+    response = str(input(prompt_name))
+    while response == 'list':
+        for names_in_list in donor_dict:
+            print(names_in_list[0])
+            
+        response = str(input(prompt_name))
+        
+    for names_in_list in donor_dict:
+        if names_in_list[0] == response:
+            # Request the user for a donation amount.
+            response_amount = input(prompt_amount)
+            x = get_index(response) 
+            donor_dict[x].append(float(response_amount))
+            break
+    
+    else:
+        # Add the name to the list
+        add_donor(response)
+        # Request the user for a donation amount.
+        response_amount = input(prompt_amount)
+        x = get_index(response)
+        donor_dict[x].append(float(response_amount))
```

Comment:
there is some duplicate code in the for loop and else block, could it be condensed ?